### PR TITLE
feat(shared): BlockInjectButton - handle onDrop validation

### PR DESCRIPTION
**Pull request by celine-s**

### CODE REVIEW

File: packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx [hunk 1]
```diff
@@ -12,6 +13,17 @@ import React from 'react';
 import { DragEventHandler, MouseEventHandler, useRef, useState } from 'react';
 import { joinClassNames } from '../../utilities/react/joinClassNames';
 import { BlockInjectButtonProps } from './types';
+import { FileExtension, FileExtensionSets } from '@frontify/app-bridge';
+
+const horizontalGroupStyles =
+    '[&:not(:first-child)]:tw-border-l-0  first:tw-rounded-tl first:tw-rounded-bl last:tw-rounded-tr last:tw-rounded-br';
+const verticalGroupStyles =
+    '[&:not(:first-child)]:tw-border-t-0  first:tw-rounded-tl first:tw-rounded-tr last:tw-rounded-br last:tw-rounded-bl';
```
Comment (ragi96) [inline packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx:21]:
```suggestion
const horizontalGroupStyles = '[&:not(:first-child)]:tw-border-l-0  first:tw-rounded-bl last:tw-rounded-tr';
const verticalGroupStyles = '[&:not(:first-child)]:tw-border-t-0  first:tw-rounded-tr last:tw-rounded-bl';
```

removed duplicated classes added within another suggestion directly as they are used anyways

Add these two "removed" classes `first:tw-rounded-tl last:tw-rounded-br` directly to the button on line 87

File: packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx [hunk 2]
```diff
@@ -24,16 +36,38 @@ export const BlockInjectButton = ({
...
+    const isValidAsset = (files: FileList) => {
+        if (!validFileType) {
+            return true;
+        }
+        for (let i = 0; i < files.length; i++) {
+            const droppedFileExtension = (files[i].name.split('.').pop() ?? '') as FileExtension;
+            if (!FileExtensionSets[validFileType].includes(droppedFileExtension)) {
+                return false;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (celine-s) [reply packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx:69]: This is not possible as we cannot iterate over files of type FileList. see [change here](https://github.com/Frontify/guideline-blocks/pull/539/commits/a21cd56cec5f95b8f93e6db7a0f82f6900c5964a) and the [type issue here](https://github.com/Frontify/guideline-blocks/actions/runs/4723203884/jobs/8378845728)

File: packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx (hunk 1 above)
Comment (celine-s) [reply packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx]: changed in [fcc9555](https://github.com/Frontify/guideline-blocks/pull/539/commits/fcc955505aa583ab7cccc1ebaa775121e52ea118). Did you mean it like this?

File: packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx [hunk 3]
```diff
@@ -12,6 +13,17 @@ import React from 'react';
 import { DragEventHandler, MouseEventHandler, useRef, useState } from 'react';
```
[the remark this answers is not in the document]
Comment (celine-s) [reply packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx]: changed 👍

File: packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx:63]: ah i remember yes.

File: packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx (hunk 1 above)
Comment (ragi96) [reply packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx]: no, these 2 classes: `first:tw-rounded-tl last:tw-rounded-br` are used as verticalStyle and horizontalStyle so they are not related to vertical or horizontal so they are always on the button and the rest is conditional, hit me up if there is further unclarity

Comment (celine-s) [reply packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx]: Fair enough. I thought it's more readable if we keep all rounded tailwind classes within one condition, but you're right. Changed it

**Review by fulopdaniel — APPROVED** [review]

**Comment by fulopdaniel** [comment]
Thanks, this was much needed

**Review by ragi96 — APPROVED** [review]

File: packages/shared/src/components/BlockInjectButton/BlockInjectButton.spec.ct.tsx [hunk 4]
```diff
@@ -23,7 +23,7 @@ describe('Block Inject Button', () => {
         );
         cy.get(BlockInjectButtonSelector)
             .should('exist')
-            .should('have.class', 'first:tw-rounded-tl first:tw-rounded-bl last:tw-rounded-tr last:tw-rounded-br');
+            .should('have.class', 'first:tw-rounded-tl last:tw-rounded-br', 'first:tw-rounded-bl last:tw-rounded-tr');
```
Comment (SamuelAlev) [inline packages/shared/src/components/BlockInjectButton/BlockInjectButton.spec.ct.tsx:26]: This is not scalable, the `have.class` is actually not necessary

File: packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx [hunk 5]
```diff
@@ -46,29 +74,49 @@ export const BlockInjectButton = ({
...
-                    : 'tw-bg-blank-state-shaded-inverse tw-border-blank-state-line tw-text-blank-state-shaded ',
+                    ? 'tw-text-blank-state-pressed tw-bg-blank-state-pressed-inverse tw-border-blank-state-line-hover'
+                    : getBorderClassNames(errorMsg),
                 fillParentContainer ? 'tw-h-full' : 'tw-h-[72px]',
-                isDraggingOver && !isLoading ? 'tw-border-dashed' : 'tw-border-solid',
+                isDraggingOver && !errorMsg && !isLoading ? 'tw-border-dashed' : 'tw-border-solid',
             ])}
-            onDragEnter={onDrop ? () => setIsDraggingOver(true) : undefined}
-            onDragLeave={onDrop ? () => setIsDraggingOver(false) : undefined}
+            onDragEnter={
```
Comment (SamuelAlev) [inline packages/shared/src/components/BlockInjectButton/BlockInjectButton.tsx:95]: There are actually no tests for the changes in this PR, please be careful!

**Pull request merged**